src/main/java/com/minecolonies/coremod/colony/requestsystem/data/StandardRequestResolverRequestAssignmentDataStore.java

```diff
@@ -0,0 +1,127 @@
+package com.minecolonies.coremod.colony.requestsystem.data;
+
+import com.google.common.reflect.TypeToken;
+import com.minecolonies.api.colony.requestsystem.StandardFactoryController;
+import com.minecolonies.api.colony.requestsystem.data.IProviderResolverAssignmentDataStore;
+import com.minecolonies.api.colony.requestsystem.data.IRequestResolverRequestAssignmentDataStore;
+import com.minecolonies.api.colony.requestsystem.factory.FactoryVoidInput;
+import com.minecolonies.api.colony.requestsystem.factory.IFactory;
+import com.minecolonies.api.colony.requestsystem.factory.IFactoryController;
+import com.minecolonies.api.colony.requestsystem.token.IToken;
+import com.minecolonies.api.util.NBTUtils;
+import com.minecolonies.api.util.constant.NbtTagConstants;
+import com.minecolonies.api.util.constant.TypeConstants;
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraft.util.Tuple;
+import net.minecraftforge.common.util.Constants;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class StandardRequestResolverRequestAssignmentDataStore implements IRequestResolverRequestAssignmentDataStore
+{
+
+    private final Map<IToken<?>, Collection<IToken<?>>> assignments;
+    private IToken<?> id;
+
+    public StandardRequestResolverRequestAssignmentDataStore(
+      final IToken<?> id,
+      final Map<IToken<?>, Collection<IToken<?>>> assignments
+      ) {
+        this.id = id;
+        this.assignments = assignments;
+    }
+
+    public StandardRequestResolverRequestAssignmentDataStore()
+    {
+        this(StandardFactoryController.getInstance().getNewInstance(TypeConstants.ITOKEN), new HashMap<>());
+    }
+
+    @NotNull
+    @Override
+    public Map<IToken<?>, Collection<IToken<?>>> getAssignments()
+    {
+        return assignments;
+    }
+
+    @Override
+    public IToken<?> getId()
+    {
+        return id;
+    }
+
+    @Override
+    public void setId(final IToken<?> id)
+    {
+        this.id = id;
+    }
+
+    public static class Factory implements IFactory<FactoryVoidInput, StandardRequestResolverRequestAssignmentDataStore>
+    {
+
+        @NotNull
+        @Override
+        public TypeToken<? extends StandardRequestResolverRequestAssignmentDataStore> getFactoryOutputType()
+        {
+            return TypeToken.of(StandardRequestResolverRequestAssignmentDataStore.class);
+        }
+
+        @NotNull
+        @Override
+        public TypeToken<? extends FactoryVoidInput> getFactoryInputType()
+        {
+            return TypeConstants.FACTORYVOIDINPUT;
+        }
+
+        @NotNull
+        @Override
+        public StandardRequestResolverRequestAssignmentDataStore getNewInstance(
+          @NotNull final IFactoryController factoryController, @NotNull final FactoryVoidInput factoryVoidInput, @NotNull final Object... context) throws IllegalArgumentException
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the declaration of thrown exception 'java.lang.IllegalArgumentException' which is a runtime exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)